File: arrays/twoSum.go
package arrays

func TowSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		p := target - nums[i];
		_, ok := m[p]
		if ok {
			return []int{ m[p], i }
		}
		m[nums[i]] = i
	}
	return []int{0, 0}
}

File: arrays/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let map = {};
    
    for(let i = 0; i < nums.length; i++) {
        let p = target - nums[i];
        if(p in map) {
            return [map[p], i];
        }
        map[nums[i]] = i;
    }
};